File: shared/src/main/java/ru/privatenull/pnauth/event/CancellableAuthEvent.java
package ru.privatenull.pnauth.event;
import ru.privatenull.pnauth.kernel.event.DecisionEvent;
public interface CancellableAuthEvent extends AuthEvent, DecisionEvent {
    default String cancellationReason() { return effectiveDecision().reason(); }
    default void cancel(String reason) { setDecision(true, reason); }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/event/CancellableAuthEvent.kt
package ru.privatenull.pnauth.event

import ru.privatenull.pnauth.kernel.event.DecisionEvent

interface CancellableAuthEvent : AuthEvent, DecisionEvent {
    fun cancellationReason(): String? = effectiveDecision().reason
    fun cancel(reason: String?) {
        setDecision(true, reason)
    }
}
